carla/recourse_methods/catalog/cem/model.py

```diff
@@ -118,14 +72,11 @@ def __init__(self, sess, catalog_model: MLModel, hyperparams):
         delta_img_s = self.orig_img - self.adv_img_s
 
         # distance to the input data
-        L2_dist = tf.reduce_sum(tf.square(delta_img), [1])
-        L2_dist_s = tf.reduce_sum(tf.square(delta_img_s), [1])
-        L1_dist = tf.reduce_sum(tf.abs(delta_img), [1])
+        L1_dist, L2_dist = self.__compute_l_norm(delta_img)
+        _, L2_dist_s = self.__compute_l_norm(delta_img_s)
 
-        enforce_input = delta_img if self.mode == "PP" else self.adv_img
-        enforce_input_s = delta_img_s if self.mode == "PP" else self.adv_img_s
-        self.ImgToEnforceLabel_Score = catalog_model.raw_model(enforce_input)
-        ImgToEnforceLabel_Score_s = catalog_model.raw_model(enforce_input_s)
+        self.ImgToEnforceLabel_Score = self.__get_label_score(delta_img, self.adv_img)
+        ImgToEnforceLabel_Score_s = self.__get_label_score(delta_img_s, self.adv_img_s)
```

Comment:
Few things still remain a bit unclear:
- What is the img_s and the corresponding score_s? Can we have clearer naming?
- Why do we need to set these scores as class members? Can't they just be function inputs/outputs? -> Same with the losses below
- Ideally we follow python naming convention (lower cased, with underscores)
- Why are the names all related to images? (img) Does that mean we cannot use this method for other datasets?

Reply:
Related to point 4: the method was originally only applied to images I think, but it works for other datasets as well.